### DISCUSSION TITLE
How to split a large graph into subgraph for training and inference

### DISCUSSION THREAD

**Discussion by OswaldoBornemann** [pyg-team/pytorch_geometric#4803] [Q&A]
Hi, May i ask how to split a large graph into subgraph for training and inference? I have noticed that can use NeighborSampler and GraphSAINTRandomWalkSampler, but i want the subgraph will not change in each training epoch. 

Really hope anyone could give me a hint, thanks.

**Comment by wsad1** [comment]
[torch_geometric.utils.subgraph](https://pytorch-geometric.readthedocs.io/en/latest/modules/utils.html#torch_geometric.utils.subgraph) returns an induced subgraph, this might be what you are looking for.

**Reply by OswaldoBornemann** [reply]
Thank you. That's great help.

**Comment by rusty1s** [comment]
Duplicate of https://github.com/pyg-team/pytorch_geometric/issues/4804
